p2p/src/peer_manager/mod.rs

```diff
@@ -495,25 +332,273 @@ where
         &mut self,
         peer_id: PeerId,
         response: Option<oneshot_nofail::Sender<crate::Result<()>>>,
-    ) -> crate::Result<()> {
+    ) {
         log::debug!("disconnect peer {peer_id}");
-
         let res = self.try_disconnect(peer_id);
 
         match res {
             Ok(()) => {
-                self.pending_disconnects.insert(peer_id, response);
+                let old_value = self.pending_disconnects.insert(peer_id, response);
+                assert!(old_value.is_none());
             }
             Err(e) => {
+                log::debug!("disconnecting new peer {peer_id} failed: {e}");
                 if let Some(response) = response {
                     response.send(Err(e));
                 }
             }
         }
+    }
+
+    /// Check if the (inbound or outbound) peer connection can be accepted
```

Comment:
Please give context to this, like an example on when it's not accepted.

Reply:
Sure, done.